lib/db.js

```diff
@@ -540,52 +531,45 @@ module.exports = (
     )
   }
 
-  DB.prototype.updateSessionToken = function (token, ip, geo) {
-    log.trace({ op: 'DB.updateSessionToken', uid: token && token.uid })
+  DB.prototype.updateSessionToken = function (token, geo) {
+    const { id, uid } = token
+
+    log.trace({ op: 'DB.updateSessionToken', id, uid })
 
-    const uid = token.uid
-    if (! this.redis || ! features.isLastAccessTimeEnabledForUser(uid)) {
+    if (! redis || ! features.isLastAccessTimeEnabledForUser(uid)) {
```

Comment:
You kind of mentioned it below, but it was a little confusing seeing ` if (! redis)` and `if (! this.redis)` in different parts of the code. Having a helper function to check whether it is enabled and online might be helpful.